static/js/src/advantage/react/hooks/useLastPurchaseIds.ts

```diff
@@ -0,0 +1,25 @@
+import { getLastPurchaseIds } from "advantage/api/contracts";
+import { LastPurchaseIds, UserSubscription } from "advantage/api/types";
+import { useQuery, UseQueryOptions } from "react-query";
+
+/**
+ * Get a the last purchase id by account id and period.
+ */
+export const selectPurchaseIdsByPeriod = (
```

Comment:
...but a more explicit name would remove the need for a comment

```suggestion
export const selectLastPurchaseIdByAccountAndPeriod = (
```